test/e2e/apps/deployment.go

```diff
@@ -408,16 +409,17 @@ var _ = SIGDescribe("Deployment", func() {
 			if deployment, ok := event.Object.(*appsv1.Deployment); ok {
 				found := deployment.ObjectMeta.Name == testDeployment.Name &&
 					deployment.ObjectMeta.Labels["test-deployment-static"] == "true" &&
-					deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas &&
+					deployment.Status.ReadyReplicas == testDeploymentMinimumReplicas &&
```

Comment:
@oomichi We were looking for the wrong number of replicas here, hence it passing often, and flaking once in awhile. This is the updated correct logic.

Reply:
I still have a question here.
testDeploymentDefaultReplicas is 2 and testDeploymentMinimumReplicas is 1 on the code.
That means now deployment.Status.ReadyReplicas is 2 in most test runs.
If we change here to testDeploymentMinimumReplicas(1), `found` doesn't become true in the case of testDeploymentDefaultReplicas(2). That seems still flake.

If we are thinking testDeploymentMinimumReplicas also can happen, why don't we just add the condition like
```
(deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas ||
 deployment.Status.ReadyReplicas == testDeploymentMinimumReplicas) &&
```
?

Reply:
Using a section from `ci-kubernetes-e2e-gci-gce/1326774472585252864/build-log.txt` as an example, it shows the deployment has scaled down from 2 replicas to 1. From what I see, when it's checking after "fetching" it always finds the first result with 1 replica, which should be the default check IMO. When it doesn't it's working through a list and I think it finds one from when the deployment was at 2 replicas. My guess for why the test flaked is that the earlier watch had been discarded for some reason which meant there wasn't a match to find causing the test to fail.

Does this help explain my current logic @oomichi or is there something I'm overlooking? TIA

```
...
I1112 06:42:27.058] Nov 12 06:42:21.878: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 2
I1112 06:42:27.058] Nov 12 06:42:21.878: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 2
I1112 06:42:27.059] Nov 12 06:42:21.894: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 1
I1112 06:42:27.059] [1mSTEP[0m: listing Deployments
I1112 06:42:27.059] Nov 12 06:42:21.949: INFO: Found test-deployment with labels: map[test-deployment:patched test-deployment-static:true]
I1112 06:42:27.059] [1mSTEP[0m: updating the Deployment
I1112 06:42:27.059] Nov 12 06:42:22.055: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 1
I1112 06:42:27.059] [1mSTEP[0m: fetching the DeploymentStatus
I1112 06:42:27.059] Nov 12 06:42:22.158: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 1 and labels map[test-deployment:updated test-deployment-static:true]
I1112 06:42:27.060] Nov 12 06:42:22.158: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 1 and labels map[test-deployment:updated test-deployment-static:true]
I1112 06:42:27.060] Nov 12 06:42:22.158: INFO: observed Deployment test-deployment in namespace deployment-5460 with ReadyReplicas 1 and labels map[test-deployment:updated test-deployment-static:true]
...
```

Reply:
Here, similarly to previous comment, we expect 2 and we should reach 2 replicas. That can't change.